snmp/datadog_checks/snmp/snmp.py

```diff
@@ -72,6 +72,8 @@ def _load_conf(self, instance):
         tags = instance.get("tags", [])
         ip_address = instance["ip_address"]
         metrics = instance.get('metrics', [])
+        if self.init_config is not None and self.init_config.get('metrics') is not None:
```

Comment:
you want to do this the other way around. Here metrics defined in init_config will override instance metrics.